### PR TITLE
MINIFI-43 Fixing provenance instantiation and making some default con…

### PR DESCRIPTION
**Pull request by JPercivall**
…structors private to prevent same problem in other classes

I made the no parameter constructor private when there are required properties with no default value.

### CODE REVIEW

**Comment by apiri** [comment]
Reviewing.

File: minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/schema/ConfigSchema.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
 
     private ProvenanceRepositorySchema provenanceRepositorySchema;
 
-    public ConfigSchema() {
+    private ConfigSchema() {
```
Comment (apiri) [inline minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/schema/ConfigSchema.java]: Changing to private, should this c'tor just be removed?

File: minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/schema/ConnectionSchema.java [hunk 2]
```diff
@@ -44,7 +44,7 @@
     private String flowfileExpiration = "0 sec";
     private String queuePrioritizerClass = "";
 
-    public ConnectionSchema() {
+    private ConnectionSchema() {
```
Comment (apiri) [inline minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/schema/ConnectionSchema.java]: Same as above

**Comment by apiri** [comment]
Hey @JPercivall.

Changes look good to resolve the listed issue in the ticket.  Aside from that, I think we can probably clean up the private constructors as it appears we have a defined constructor for each using the map.

Also, would you be able to cleanup https://github.com/JPercivall/nifi-minifi/blob/1c5c2d49e5e0477a0b6dbb58cbd2c3ecf73b4a52/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/schema/ConfigSchema.java#L59?  I realize it wasn't added by this commit, but might be nice to remove it while making changes.

**Comment by JPercivall** [comment]
@apiri I agree with your comments. I was looking for the easy fix (just make them private) but didn't think it all the way through. Thanks

I made the adjustment and removed the extra object in ConfigSchema

**Comment by apiri** [comment]
@JPercivall looks good.  Rock and roll :guitar: +1

**Comment by JPercivall** [comment]
Thanks @apiri I merged it into master

**Pull request closed without merging**